### ISSUE TITLE
Error when running pip install pyhcl

### ISSUE THREAD

**Issue by gposton** [virtuald/pyhcl#3]
This seems innate as I'm able to use the package despite the install failure, but I do get this error:

    $ pip install pyhcl
    Collecting pyhcl
      Using cached pyhcl-0.1.11.tar.gz
    Requirement already satisfied (use --upgrade to upgrade): ply==3.4 in /Users/glennposton/.virtualenvs/skippy/lib/python3.4/site-packages (from pyhcl)
    Building wheels for collected packages: pyhcl
      Running setup.py bdist_wheel for pyhcl
      Complete output from command /Users/glennposton/.virtualenvs/skippy/bin/python3.4 -c "import setuptools;__file__='/private/var/folders/zr/mq8gh7k912324dntrd_sr_0m0000gn/T/pip-build-u4jkjf67/pyhcl/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /var/folders/zr/mq8gh7k912324dntrd_sr_0m0000gn/T/tmp3z0z3qbtpip-wheel-:
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib
      creating build/lib/hcl
      copying src/hcl/__init__.py -> build/lib/hcl
      copying src/hcl/api.py -> build/lib/hcl
      copying src/hcl/lexer.py -> build/lib/hcl
      copying src/hcl/parser.py -> build/lib/hcl
      copying src/hcl/version.py -> build/lib/hcl
      running build_scripts
      creating build/scripts-3.4
      copying and adjusting scripts/hcltool -> build/scripts-3.4
      changing mode of build/scripts-3.4/hcltool from 644 to 755
      installing to build/bdist.macosx-10.6-intel/wheel
      running install
      running install_lib
      creating build/bdist.macosx-10.6-intel
      creating build/bdist.macosx-10.6-intel/wheel
      creating build/bdist.macosx-10.6-intel/wheel/hcl
      copying build/lib/hcl/__init__.py -> build/bdist.macosx-10.6-intel/wheel/hcl
      copying build/lib/hcl/api.py -> build/bdist.macosx-10.6-intel/wheel/hcl
      copying build/lib/hcl/lexer.py -> build/bdist.macosx-10.6-intel/wheel/hcl
      copying build/lib/hcl/parser.py -> build/bdist.macosx-10.6-intel/wheel/hcl
      copying build/lib/hcl/version.py -> build/bdist.macosx-10.6-intel/wheel/hcl
      running install_egg_info
      running egg_info
      writing requirements to src/pyhcl.egg-info/requires.txt
      writing top-level names to src/pyhcl.egg-info/top_level.txt
      writing dependency_links to src/pyhcl.egg-info/dependency_links.txt
      writing src/pyhcl.egg-info/PKG-INFO
      warning: manifest_maker: standard file '-c' not found

      reading manifest file 'src/pyhcl.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      writing manifest file 'src/pyhcl.egg-info/SOURCES.txt'
      Copying src/pyhcl.egg-info to build/bdist.macosx-10.6-intel/wheel/pyhcl-0.1.11-py3.4.egg-info
      running install_scripts
      creating build/bdist.macosx-10.6-intel/wheel/pyhcl-0.1.11.data
      creating build/bdist.macosx-10.6-intel/wheel/pyhcl-0.1.11.data/scripts
      copying build/scripts-3.4/hcltool -> build/bdist.macosx-10.6-intel/wheel/pyhcl-0.1.11.data/scripts
      changing mode of build/bdist.macosx-10.6-intel/wheel/pyhcl-0.1.11.data/scripts/hcltool to 755
      Generating parse table...
      Traceback (most recent call last):
        File "<string>", line 1, in <module>
        File "/private/var/folders/zr/mq8gh7k912324dntrd_sr_0m0000gn/T/pip-build-u4jkjf67/pyhcl/setup.py", line 79, in <module>
          "Topic :: Text Processing"
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/core.py", line 148, in setup
          dist.run_commands()
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 955, in run_commands
          self.run_command(cmd)
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
          cmd_obj.run()
        File "/Users/glennposton/.virtualenvs/skippy/lib/python3.4/site-packages/wheel/bdist_wheel.py", line 211, in run
          self.run_command('install')
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
          cmd_obj.run()
        File "/private/var/folders/zr/mq8gh7k912324dntrd_sr_0m0000gn/T/pip-build-u4jkjf67/pyhcl/setup.py", line 29, in run
          self.execute(_post_install, (), msg="Generating parse table...")
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/cmd.py", line 336, in execute
          util.execute(func, args, msg, dry_run=self.dry_run)
        File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/util.py", line 301, in execute
          func(*args)
        File "/private/var/folders/zr/mq8gh7k912324dntrd_sr_0m0000gn/T/pip-build-u4jkjf67/pyhcl/setup.py", line 21, in _post_install
          import hcl
      ImportError: No module named 'hcl'

      ----------------------------------------
      Failed building wheel for pyhcl
    Failed to build pyhcl
    Installing collected packages: pyhcl
      Running setup.py install for pyhcl
    Successfully installed pyhcl-0.1.11

Running on OSX 10.10.5 with the following python versions:
    
    $ python --version
    Python 3.4.3
    $ virtualenv --version
    13.1.2
    $ pip --version
    pip 7.1.2 from /Users/glennposton/.virtualenvs/skippy/lib/python3.4/site-packages (python 3.4)

**Comment by scalp42** [comment]
I'm having the same issue:

```
Python 2.7.10
pip 7.1.2
```

**Comment by virtuald** [comment]
I'll look at it this weekend.

**Comment by virtuald** [comment]
It appears the bug only shows up when you have an updated version of setuptools installed, I played with it a bunch and determined that moving the parsetab.dat generation to a different place should fix it.

**Comment by scalp42** [comment]
@virtuald thanks for the comment, unfortunately I'm seeing a new issue:

```
$> pip install pyhcl --upgrade                                                                             

Collecting pyhcl
  Using cached pyhcl-0.1.13.tar.gz
Requirement already up-to-date: ply==3.4 in /usr/local/lib/python2.7/site-packages (from pyhcl)
Building wheels for collected packages: pyhcl
  Running setup.py bdist_wheel for pyhcl
  Complete output from command /usr/local/opt/python/bin/python2.7 -c "import setuptools;__file__='/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/tmp_AnIaGpip-wheel-:
  running bdist_wheel
  running build
  running build_py
  Generating parse table...
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 92, in <module>
      "Topic :: Text Processing"
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
      dist.run_commands()
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
      self.run_command(cmd)
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python2.7/site-packages/wheel/bdist_wheel.py", line 175, in run
      self.run_command('build')
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
      self.distribution.run_command(command)
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
      cmd_obj.run()
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/command/build.py", line 127, in run
      self.run_command(cmd_name)
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
      self.distribution.run_command(command)
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
      cmd_obj.run()
    File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 38, in run
      self.execute(_pre_install, (), msg="Generating parse table...")
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 349, in execute
      util.execute(func, args, msg, dry_run=self.dry_run)
    File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/util.py", line 309, in execute
      func(*args)
    File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 25, in _pre_install
      os.unlink(dat)
  NameError: global name 'os' is not defined

  ----------------------------------------
  Failed building wheel for pyhcl
Failed to build pyhcl
Installing collected packages: pyhcl
  Found existing installation: pyhcl 0.1.11
    Uninstalling pyhcl-0.1.11:
      Successfully uninstalled pyhcl-0.1.11
  Running setup.py install for pyhcl
    Complete output from command /usr/local/opt/python/bin/python2.7 -c "import setuptools, tokenize;__file__='/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-aV6oJi-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    Generating parse table...
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 92, in <module>
        "Topic :: Text Processing"
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/lib/python2.7/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/command/install.py", line 563, in run
        self.run_command('build')
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/command/build.py", line 127, in run
        self.run_command(cmd_name)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 38, in run
        self.execute(_pre_install, (), msg="Generating parse table...")
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/cmd.py", line 349, in execute
        util.execute(func, args, msg, dry_run=self.dry_run)
      File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/distutils/util.py", line 309, in execute
        func(*args)
      File "/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py", line 25, in _pre_install
        os.unlink(dat)
    NameError: global name 'os' is not defined

    ----------------------------------------
  Rolling back uninstall of pyhcl
Command "/usr/local/opt/python/bin/python2.7 -c "import setuptools, tokenize;__file__='/private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-aV6oJi-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /private/var/folders/p_/_jy68n853bx7gj0k50fcsjjr0000gn/T/pip-build-31LcS8/pyhcl
```

**Comment by virtuald** [comment]
Sorry about that. Fixed in 0.1.14

**Comment by scalp42** [comment]
Thanks a lot @virtuald for the support fast response :smile:
